packages/core/lib/commands/test/run.js

```diff
@@ -68,43 +86,50 @@ module.exports = async function (options) {
     inputFile: file
   });
 
-  if (config.networks[config.network]) {
-    await Environment.detect(config);
-    const { temporaryDirectory } = await copyArtifactsToTempDir(config);
-    const numberOfFailures = await prepareConfigAndRunTests({
-      config,
-      files,
-      temporaryDirectory
-    });
-    return numberOfFailures;
-  } else {
-    const ipcOptions = { network: "test" };
-    const port = await require("get-port")();
+  const configuredNetwork = config.networks[config.network];
+  const testNetworkDefined = configuredNetwork && config.network === "test";
+  const noProviderHostOrUrlConfigured =
+    configuredNetwork &&
+    !configuredNetwork.provider &&
+    !configuredNetwork.host &&
+    !configuredNetwork.url;
+  const ipcOptions = { network: "test" };
+  let numberOfFailures, port;
+  let ganacheOptions = {
+    host: "127.0.0.1",
+    port,
+    network_id: 4447,
+    mnemonic:
+      "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat",
+    time: config.genesis_time,
+    miner: {
+      instamine: "strict"
+    }
+  };
 
-    const ganacheOptions = {
-      host: "127.0.0.1",
-      port,
-      network_id: 4447,
-      mnemonic:
-        "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat",
-      time: config.genesis_time,
-      miner: {
-        instamine: "strict"
-      }
-    };
-    const { disconnect } = await Develop.connectOrStart(
+  if (
+    (testNetworkDefined && noProviderHostOrUrlConfigured) ||
+    !configuredNetwork
+  ) {
+    // Use managed ganache with overriding user specified config or without any specification in the config
+    port = await require("get-port")();
```

Comment:
You only use `port` in this block, so rather than declaring it with `let` up above, I think it would make more sense to declare it `const` right here.